java/src/jmri/jmrit/display/layoutEditor/ConnectivityUtil.java

```diff
@@ -119,7 +129,9 @@ public List<LayoutTrackExpectedState<LayoutTurnout>> getTurnoutList(
             @CheckForNull Block currBlock,
             @CheckForNull Block prevBlock,
             @CheckForNull Block nextBlock,
-            boolean suppress) {
+            boolean suppress,
+            AtomicBoolean notFound1 ) {
+        notFound1.set(false);  //notfound1 is set true if notfound is true later in code and returned to calling routine
```

Comment:
I might be wrong, but I believe this change does not do what is expected.

The only thing returned is an arrayList of LayoutTrackExpectedState<LayoutTurnout> objects, which might be empty which implies nothing found.

It appears that the intention is to have an extended search that crosses panel boundaries.  If so, the following changes would be needed.

Line 106 is removed.
Line 107 `return getTurnoutList(currBlock, prevBlock, nextBlock, suppress, false);`
Lines 133 `boolean extendedSearch);`
Line 134 is removed.

The extended search logic is wrapped in `if (extendedSearch) {???}`

The existing calls at 93 and 101 must behave the same as before.

Reply:
I agree the code looks a bit stupid in the ConnectivityUtil, especially in 
    public List<LayoutTrackExpectedState<LayoutTurnout>> getTurnoutList(
            @CheckForNull Block currBlock,
            @CheckForNull Block prevBlock,
            @CheckForNull Block nextBlock,
            boolean suppress) {
        AtomicBoolean notFound1 = new AtomicBoolean(true);
        return getTurnoutList(currBlock, prevBlock, nextBlock, suppress, notFound1);
    }
where an atomic boolean is defined and not used. 

The whole point of the atomic boolean was to do some logging and breaking the code in a convenient fashion in AutoTurnouts when I was developing the code line 287 onwards.

Maybe there is a better way of doing things without the Atomic Boolean, but I am not a very good programmer, and that was the easiest thing to do at the time.

Reply:
I tried doing the test 
"The only thing returned is an arrayList of LayoutTrackExpectedState objects, which might be empty which implies nothing found."
in AutoTurnouts and got into problems, and hit on the idea of an Atomic Boolean. Maybe I should try again...